### ISSUE TITLE
No test examples

### ISSUE THREAD

**Issue by jdrago999** [level09/enferno#3]
The website enferno.io does not even contain the word "test"

https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=site%3Aenferno.io%20test

How does one go about testing an enferno application?

**Comment by level09** [comment]
Thanks for your interest in this project. 

Testing is a big topic, I'm not sure which test are you referring to, functional/unit testing ? load testing? security/usability/etc .. ? 

depending on the kind of test you are referring to, perhaps the best way to test it is to just download it and run it. 

if you think some unit/functional tests can be helpful, I think I can add that. 

Feel free to submit any pull requests as well :)

**Comment by jdrago999** [comment]
Hi - ideally if I'm going to write something, I prefer to begin with tests.  In the case of a web application, I want to test my data layer, my server-side logic and my presentation layer separately.  I'd even like to have the ability to mock resources that I'm not testing, for the sake of creating an arbitrary set of circumstances in which to test the behavior of different parts of my application (eg: database lookup fails -- how does the controller handle that?).

**Comment by level09** [comment]
I agree with most what you say, and I think testing is important. but I honestly took an opinionated approach and I really have a limited time to write tests. plus Enferno is a lightweight layer sits on top of Flask framework which I think is well tested. 

This talk I attended in Europython Italy has also changed my mind about testing: 

https://ep2013.europython.eu/conference/talks/tdm-test-driven-madness

Also, if this helps, we use Enferno at a very large media corporation and it runs smoothly serving over 15M visitors/month on a single server (medium box).

**Comment by jdrago999** [comment]
Sure it runs smoothly now, but how about when the new person comes onto your team?  When they add a new feature which breaks something else, you won't know until it's already been deployed to production.  No tests == new bugs.  There's no way around it.  That's not to say that tests == no bugs, but as executable documentation they are a solid practice.

I do hope you reconsider your choice in not making testing an integral part of this framework.
